File: backend/app/schemas/marketing.py
"""Pydantic v2 schemas for the Marketing Site Builder module.

A handful of conventions:

* Every list-shaped entity gets ``Create`` / ``Update`` / ``Out`` variants.
* The settings singleton is exposed as ``MarketingSettingsOut`` /
  ``MarketingSettingsUpdate`` (no ``Create`` — it's seeded once at first boot).
* ``MarketingStateOut`` bundles everything into a single payload so the editor
  and the public renderer can hydrate from one request.
* Section payloads are typed as ``dict[str, Any]`` because shape varies by
  section type; validation happens in the editor.
"""
from __future__ import annotations

from datetime import datetime
from typing import Any, Literal

from pydantic import BaseModel, Field

from app.schemas.common import ORMModel


# ---------------------------------------------------------------------------
# Reusable types
# ---------------------------------------------------------------------------
SectionType = Literal["hero", "services", "projects", "testimonials", "about", "cta"]
PostStatus = Literal["draft", "published"]
ThemeMode = Literal["light", "dark"]
ButtonStyle = Literal["square", "rounded"]
Density = Literal["comfortable", "compact"]
NavRoute = Literal["home", "about", "services", "portfolio", "blog", "contact"]


# ---------------------------------------------------------------------------
# Settings (singleton)
# ---------------------------------------------------------------------------
class MarketingSettingsOut(ORMModel):
    key: str
    name: str
    tagline: str
    description: str
    logo_text: str
    logo_dot: bool
    base_url: str
    seo_title: str
    seo_description: str
    theme_mode: str
    primary_color: str
    accent_color: str
    heading_font: str
    body_font: str
    button_style: str
    density: str
    radius: int
    contact_email: str
    contact_phone: str
    contact_address: str
    contact_hours: str
    updated_at: datetime


class MarketingSettingsUpdate(BaseModel):
    name: str | None = Field(default=None, max_length=180)
    tagline: str | None = Field(default=None, max_length=300)
    description: str | None = None
    logo_text: str | None = Field(default=None, max_length=180)
    logo_dot: bool | None = None
    base_url: str | None = Field(default=None, max_length=500)
    seo_title: str | None = Field(default=None, max_length=300)
    seo_description: str | None = None
    theme_mode: ThemeMode | None = None
    primary_color: str | None = Field(default=None, max_length=16)
    accent_color: str | None = Field(default=None, max_length=16)
    heading_font: str | None = Field(default=None, max_length=120)
    body_font: str | None = Field(default=None, max_length=120)
    button_style: ButtonStyle | None = None
    density: Density | None = None
    radius: int | None = Field(default=None, ge=0, le=64)
    contact_email: str | None = Field(default=None, max_length=255)
    contact_phone: str | None = Field(default=None, max_length=60)
    contact_address: str | None = Field(default=None, max_length=500)
    contact_hours: str | None = Field(default=None, max_length=255)


# ---------------------------------------------------------------------------
# Navigation
# ---------------------------------------------------------------------------
class NavItemIn(BaseModel):
    """One nav row — used in the bulk PUT /navigation payload."""

    id: str | None = None
    label: str = Field(min_length=1, max_length=120)
    route: NavRoute
    enabled: bool = True
    order: int = 0


class NavItemOut(ORMModel):
    id: str
    label: str
    route: str
    enabled: bool
    order: int


# ---------------------------------------------------------------------------
# Sections
# ---------------------------------------------------------------------------
class SectionCreate(BaseModel):
    type: SectionType
    enabled: bool = True
    eyebrow: str = Field(default="", max_length=180)
    title: str = Field(default="", max_length=300)
    body: str = ""
    order: int = 0
    payload: dict[str, Any] = Field(default_factory=dict)


class SectionUpdate(BaseModel):
    type: SectionType | None = None
    enabled: bool | None = None
    eyebrow: str | None = Field(default=None, max_length=180)
    title: str | None = Field(default=None, max_length=300)
    body: str | None = None
    order: int | None = None
    payload: dict[str, Any] | None = None


class SectionOut(ORMModel):
    id: str
    type: str
    enabled: bool
    eyebrow: str
    title: str
    body: str
    order: int
    payload: dict[str, Any]
    created_at: datetime
    updated_at: datetime


# ---------------------------------------------------------------------------
# Projects
# ---------------------------------------------------------------------------
class ProjectCreate(BaseModel):
    title: str = Field(min_length=1, max_length=300)
    client: str = Field(default="", max_length=180)
    category: str = Field(default="", max_length=120)
    summary: str = ""
    body: str = ""
    year: str = Field(default="", max_length=16)
    tags: list[str] = Field(default_factory=list)
    featured: bool = False
    image_id: str | None = None
    external_url: str = Field(default="", max_length=500)
    order: int = 0


class ProjectUpdate(BaseModel):
    title: str | None = Field(default=None, min_length=1, max_length=300)
    client: str | None = Field(default=None, max_length=180)
    category: str | None = Field(default=None, max_length=120)
    summary: str | None = None
    body: str | None = None
    year: str | None = Field(default=None, max_length=16)
    tags: list[str] | None = None
    featured: bool | None = None
    image_id: str | None = None
    external_url: str | None = Field(default=None, max_length=500)
    order: int | None = None


class ProjectOut(ORMModel):
    id: str
    title: str
    slug: str
    client: str
    category: str
    summary: str
    body: str
    year: str
    tags: list[str]
    featured: bool
    image_id: str | None
    external_url: str
    order: int
    created_at: datetime
    updated_at: datetime


# ---------------------------------------------------------------------------
# Posts
# ---------------------------------------------------------------------------
class PostCreate(BaseModel):
    title: str = Field(min_length=1, max_length=300)
    excerpt: str = ""
    body: str = ""
    author: str = Field(default="", max_length=180)
    category: str = Field(default="", max_length=120)
    tags: list[str] = Field(default_factory=list)
    publish_date: datetime | None = None
    status: PostStatus = "draft"
    seo_title: str = Field(default="", max_length=300)
    seo_description: str = ""


class PostUpdate(BaseModel):
    title: str | None = Field(default=None, min_length=1, max_length=300)
    excerpt: str | None = None
    body: str | None = None
    author: str | None = Field(default=None, max_length=180)
    category: str | None = Field(default=None, max_length=120)
    tags: list[str] | None = None
    publish_date: datetime | None = None
    status: PostStatus | None = None
    seo_title: str | None = Field(default=None, max_length=300)
    seo_description: str | None = None


class PostOut(ORMModel):
    id: str
    title: str
    slug: str
    excerpt: str
    body: str
    author: str
    category: str
    tags: list[str]
    publish_date: datetime | None
    status: str
    seo_title: str
    seo_description: str
    created_at: datetime
    updated_at: datetime


# ---------------------------------------------------------------------------
# Services
# ---------------------------------------------------------------------------
class ServiceCreate(BaseModel):
    icon: str = Field(default="", max_length=60)
    title: str = Field(min_length=1, max_length=180)
    summary: str = ""
    details: str = ""
    price: str = Field(default="", max_length=120)
    featured: bool = False
    order: int = 0


class ServiceUpdate(BaseModel):
    icon: str | None = Field(default=None, max_length=60)
    title: str | None = Field(default=None, min_length=1, max_length=180)
    summary: str | None = None
    details: str | None = None
    price: str | None = Field(default=None, max_length=120)
    featured: bool | None = None
    order: int | None = None


class ServiceOut(ORMModel):
    id: str
    icon: str
    title: str
    summary: str
    details: str
    price: str
    featured: bool
    order: int
    created_at: datetime
    updated_at: datetime


# ---------------------------------------------------------------------------
# Testimonials
# ---------------------------------------------------------------------------
class TestimonialCreate(BaseModel):
    quote: str = Field(min_length=1)
    author: str = Field(default="", max_length=180)
    role: str = Field(default="", max_length=180)
    order: int = 0


class TestimonialUpdate(BaseModel):
    quote: str | None = None
    author: str | None = Field(default=None, max_length=180)
    role: str | None = Field(default=None, max_length=180)
    order: int | None = None


class TestimonialOut(ORMModel):
    id: str
    quote: str
    author: str
    role: str
    order: int
    created_at: datetime
    updated_at: datetime


# ---------------------------------------------------------------------------
# FAQs
# ---------------------------------------------------------------------------
class FAQCreate(BaseModel):
    question: str = Field(min_length=1)
    answer: str = ""
    order: int = 0


class FAQUpdate(BaseModel):
    question: str | None = None
    answer: str | None = None
    order: int | None = None


class FAQOut(ORMModel):
    id: str
    question: str
    answer: str
    order: int
    created_at: datetime
    updated_at: datetime


# ---------------------------------------------------------------------------
# Team
# ---------------------------------------------------------------------------
class TeamMemberCreate(BaseModel):
    name: str = Field(min_length=1, max_length=180)
    role: str = Field(default="", max_length=180)
    image_id: str | None = None
    order: int = 0


class TeamMemberUpdate(BaseModel):
    name: str | None = Field(default=None, min_length=1, max_length=180)
    role: str | None = Field(default=None, max_length=180)
    image_id: str | None = None
    order: int | None = None


class TeamMemberOut(ORMModel):
    id: str
    name: str
    role: str
    image_id: str | None
    order: int
    created_at: datetime
    updated_at: datetime


# ---------------------------------------------------------------------------
# Social links
# ---------------------------------------------------------------------------
class SocialLinkCreate(BaseModel):
    platform: str = Field(min_length=1, max_length=60)
    label: str = Field(default="", max_length=180)
    url: str = Field(min_length=1, max_length=500)
    order: int = 0


class SocialLinkUpdate(BaseModel):
    platform: str | None = Field(default=None, min_length=1, max_length=60)
    label: str | None = Field(default=None, max_length=180)
    url: str | None = Field(default=None, min_length=1, max_length=500)
    order: int | None = None


class SocialLinkOut(ORMModel):
    id: str
    platform: str
    label: str
    url: str
    order: int
    created_at: datetime
    updated_at: datetime


# ---------------------------------------------------------------------------
# Media library (uploads)
# ---------------------------------------------------------------------------
class MarketingUploadOut(ORMModel):
    id: str
    filename: str
    content_type: str
    size_bytes: int
    storage_path: str
    uploaded_by_id: str | None
    created_at: datetime


# ---------------------------------------------------------------------------
# Reorder / bulk
# ---------------------------------------------------------------------------
class ReorderEntry(BaseModel):
    id: str
    order: int


# ---------------------------------------------------------------------------
# Launch checklist
# ---------------------------------------------------------------------------
class LaunchChecklistItem(BaseModel):
    id: str
    label: str
    done: bool


# ---------------------------------------------------------------------------
# Combined state — used for editor hydration + public render
# ---------------------------------------------------------------------------
class MarketingStateOut(BaseModel):
    settings: MarketingSettingsOut
    navigation: list[NavItemOut]
    sections: list[SectionOut]
    projects: list[ProjectOut]
    posts: list[PostOut]
    services: list[ServiceOut]
    testimonials: list[TestimonialOut]
    faqs: list[FAQOut]
    team: list[TeamMemberOut]
    social_links: list[SocialLinkOut]
    uploads: list[MarketingUploadOut]

File: frontend/src/lib/marketing.ts
import { api } from './api';

/* -------------------------------------------------------------------------- */
/*  Entity types — mirrored from the backend Pydantic schemas.                */
/* -------------------------------------------------------------------------- */

export type ThemeMode = 'light' | 'dark';
export type ButtonStyle = 'square' | 'rounded';
export type Density = 'comfortable' | 'compact';
export type SectionType = 'hero' | 'services' | 'projects' | 'testimonials' | 'about' | 'cta';
export type PostStatus = 'draft' | 'published';

export type MarketingSettings = {
  name: string;
  tagline: string;
  description: string;
  logoText: string;
  logoDot: boolean;
  baseUrl: string;
  seoTitle: string;
  seoDescription: string;
  themeMode: ThemeMode;
  primaryColor: string;
  accentColor: string;
  headingFont: string;
  bodyFont: string;
  buttonStyle: ButtonStyle;
  density: Density;
  radius: number;
  contactEmail: string;
  contactPhone: string;
  contactAddress: string;
  contactHours: string;
};

export type MarketingNavItem = {
  id: string;
  label: string;
  route: string;
  enabled: boolean;
  order: number;
};

export type MarketingSection = {
  id: string;
  type: SectionType;
  enabled: boolean;
  eyebrow: string;
  title: string;
  body: string;
  payload: Record<string, unknown>;
  order: number;
};

export type MarketingProject = {
  id: string;
  title: string;
  client: string;
  category: string;
  summary: string;
  body: string;
  year: string;
  tags: string[];
  featured: boolean;
  imageId: string | null;
  externalUrl: string;
};

export type MarketingPost = {
  id: string;
  title: string;
  slug: string;
  excerpt: string;
  body: string;
  author: string;
  category: string;
  tags: string[];
  publishDate: string; // ISO
  status: PostStatus;
  seoTitle: string;
  seoDescription: string;
};

export type MarketingService = {
  id: string;
  icon: string;
  title: string;
  summary: string;
  details: string;
  price: string;
  featured: boolean;
  order: number;
};

export type MarketingTestimonial = {
  id: string;
  quote: string;
  author: string;
  role: string;
  order: number;
};

export type MarketingFAQ = {
  id: string;
  question: string;
  answer: string;
  order: number;
};

export type MarketingTeamMember = {
  id: string;
  name: string;
  role: string;
  imageId: string | null;
  order: number;
};

export type MarketingSocialLink = {
  id: string;
  platform: string;
  label: string;
  url: string;
  order: number;
};

export type MarketingUpload = {
  id: string;
  filename: string;
  contentType: string;
  sizeBytes: number;
  url: string;
  createdAt: string;
};

export type MarketingChecklistItem = {
  id: string;
  label: string;
  complete: boolean;
};

export type MarketingState = {
  settings: MarketingSettings;
  navigation: MarketingNavItem[];
  sections: MarketingSection[];
  projects: MarketingProject[];
  posts: MarketingPost[];
  services: MarketingService[];
  testimonials: MarketingTestimonial[];
  faqs: MarketingFAQ[];
  team: MarketingTeamMember[];
  social_links: MarketingSocialLink[];
  contact_info: {
    email: string;
    phone: string;
    address: string;
    hours: string;
  };
};

export type MarketingStateOut = MarketingState;

export type MarketingTemplateSummary = {
  id: string;
  name: string;
  description: string;
  preview_image: string | null;
};

export type MarketingTemplate = MarketingTemplateSummary & MarketingState;

/* -------------------------------------------------------------------------- */
/*  Create/update payload aliases — backend accepts partials for PATCH.        */
/* -------------------------------------------------------------------------- */

export type SectionInput = Omit<MarketingSection, 'id'>;
export type ProjectInput = Omit<MarketingProject, 'id'>;
export type PostInput = Omit<MarketingPost, 'id'>;
export type ServiceInput = Omit<MarketingService, 'id'>;
export type TestimonialInput = Omit<MarketingTestimonial, 'id'>;
export type FAQInput = Omit<MarketingFAQ, 'id'>;
export type TeamInput = Omit<MarketingTeamMember, 'id'>;
export type SocialLinkInput = Omit<MarketingSocialLink, 'id'>;
export type NavItemInput = Omit<MarketingNavItem, 'id'>;

export type ReorderEntry = { id: string; order: number };

/* -------------------------------------------------------------------------- */
/*  Helper — some backends return camelCase, others snake_case. The wrapper   */
/*  normalises a few known fields so pages stay typed against camelCase.      */
/* -------------------------------------------------------------------------- */

type RawUpload = Partial<MarketingUpload> & {
  content_type?: string;
  size_bytes?: number;
  created_at?: string;
};

function normaliseUpload(raw: RawUpload): MarketingUpload {
  return {
    id: String(raw.id ?? ''),
    filename: raw.filename ?? '',
    contentType: raw.contentType ?? raw.content_type ?? 'application/octet-stream',
    sizeBytes: raw.sizeBytes ?? raw.size_bytes ?? 0,
    url: raw.url ?? '',
    createdAt: raw.createdAt ?? raw.created_at ?? new Date().toISOString(),
  };
}

/* -------------------------------------------------------------------------- */
/*  API surface — every endpoint in the marketing contract.                    */
/* -------------------------------------------------------------------------- */

const PREFIX = '/marketing';

export const marketingApi = {
  getState: () => api.get<MarketingState>(`${PREFIX}/state`).then((r) => r.data),

  // Settings — singleton; PATCH accepts any subset of MarketingSettings.
  updateSettings: (body: Partial<MarketingSettings>) =>
    api.patch<MarketingSettings>(`${PREFIX}/settings`, body).then((r) => r.data),

  // Navigation — atomic replace of the full nav list (server assigns IDs).
  replaceNavigation: (items: (Partial<MarketingNavItem> & { label: string; route: string })[]) =>
    api.put<MarketingNavItem[]>(`${PREFIX}/navigation`, items).then((r) => r.data),

  // Sections
  listSections: () => api.get<MarketingSection[]>(`${PREFIX}/sections`).then((r) => r.data),
  createSection: (body: Partial<SectionInput>) =>
    api.post<MarketingSection>(`${PREFIX}/sections`, body).then((r) => r.data),
  updateSection: (id: string, body: Partial<SectionInput>) =>
    api.patch<MarketingSection>(`${PREFIX}/sections/${id}`, body).then((r) => r.data),
  deleteSection: (id: string) =>
    api.delete(`${PREFIX}/sections/${id}`).then(() => true),
  reorderSections: (order: ReorderEntry[]) =>
    api.put(`${PREFIX}/sections/reorder`, order).then(() => true),

  // Projects
  listProjects: () => api.get<MarketingProject[]>(`${PREFIX}/projects`).then((r) => r.data),
  getProject: (id: string) =>
    api.get<MarketingProject>(`${PREFIX}/projects/${id}`).then((r) => r.data),
  createProject: (body: Partial<ProjectInput>) =>
    api.post<MarketingProject>(`${PREFIX}/projects`, body).then((r) => r.data),
  updateProject: (id: string, body: Partial<ProjectInput>) =>
    api.patch<MarketingProject>(`${PREFIX}/projects/${id}`, body).then((r) => r.data),
  deleteProject: (id: string) =>
    api.delete(`${PREFIX}/projects/${id}`).then(() => true),
  reorderProjects: (order: ReorderEntry[]) =>
    api.put(`${PREFIX}/projects/reorder`, order).then(() => true),

  // Posts
  listPosts: () => api.get<MarketingPost[]>(`${PREFIX}/posts`).then((r) => r.data),
  getPost: (id: string) => api.get<MarketingPost>(`${PREFIX}/posts/${id}`).then((r) => r.data),
  createPost: (body: Partial<PostInput>) =>
    api.post<MarketingPost>(`${PREFIX}/posts`, body).then((r) => r.data),
  updatePost: (id: string, body: Partial<PostInput>) =>
    api.patch<MarketingPost>(`${PREFIX}/posts/${id}`, body).then((r) => r.data),
  deletePost: (id: string) =>
    api.delete(`${PREFIX}/posts/${id}`).then(() => true),
  reorderPosts: (order: ReorderEntry[]) =>
    api.put(`${PREFIX}/posts/reorder`, order).then(() => true),

  // Services
  listServices: () => api.get<MarketingService[]>(`${PREFIX}/services`).then((r) => r.data),
  createService: (body: Partial<ServiceInput>) =>
    api.post<MarketingService>(`${PREFIX}/services`, body).then((r) => r.data),
  updateService: (id: string, body: Partial<ServiceInput>) =>
    api.patch<MarketingService>(`${PREFIX}/services/${id}`, body).then((r) => r.data),
  deleteService: (id: string) =>
    api.delete(`${PREFIX}/services/${id}`).then(() => true),
  reorderServices: (order: ReorderEntry[]) =>
    api.put(`${PREFIX}/services/reorder`, order).then(() => true),

  // Testimonials
  listTestimonials: () =>
    api.get<MarketingTestimonial[]>(`${PREFIX}/testimonials`).then((r) => r.data),
  createTestimonial: (body: Partial<TestimonialInput>) =>
    api.post<MarketingTestimonial>(`${PREFIX}/testimonials`, body).then((r) => r.data),
  updateTestimonial: (id: string, body: Partial<TestimonialInput>) =>
    api.patch<MarketingTestimonial>(`${PREFIX}/testimonials/${id}`, body).then((r) => r.data),
  deleteTestimonial: (id: string) =>
    api.delete(`${PREFIX}/testimonials/${id}`).then(() => true),
  reorderTestimonials: (order: ReorderEntry[]) =>
    api.put(`${PREFIX}/testimonials/reorder`, order).then(() => true),

  // FAQs
  listFaqs: () => api.get<MarketingFAQ[]>(`${PREFIX}/faqs`).then((r) => r.data),
  createFaq: (body: Partial<FAQInput>) =>
    api.post<MarketingFAQ>(`${PREFIX}/faqs`, body).then((r) => r.data),
  updateFaq: (id: string, body: Partial<FAQInput>) =>
    api.patch<MarketingFAQ>(`${PREFIX}/faqs/${id}`, body).then((r) => r.data),
  deleteFaq: (id: string) => api.delete(`${PREFIX}/faqs/${id}`).then(() => true),
  reorderFaqs: (order: ReorderEntry[]) =>
    api.put(`${PREFIX}/faqs/reorder`, order).then(() => true),

  // Team
  listTeam: () => api.get<MarketingTeamMember[]>(`${PREFIX}/team`).then((r) => r.data),
  createTeam: (body: Partial<TeamInput>) =>
    api.post<MarketingTeamMember>(`${PREFIX}/team`, body).then((r) => r.data),
  updateTeam: (id: string, body: Partial<TeamInput>) =>
    api.patch<MarketingTeamMember>(`${PREFIX}/team/${id}`, body).then((r) => r.data),
  deleteTeam: (id: string) => api.delete(`${PREFIX}/team/${id}`).then(() => true),
  reorderTeam: (order: ReorderEntry[]) =>
    api.put(`${PREFIX}/team/reorder`, order).then(() => true),

  // Social
  listSocial: () =>
    api.get<MarketingSocialLink[]>(`${PREFIX}/social-links`).then((r) => r.data),
  createSocial: (body: Partial<SocialLinkInput>) =>
    api.post<MarketingSocialLink>(`${PREFIX}/social-links`, body).then((r) => r.data),
  updateSocial: (id: string, body: Partial<SocialLinkInput>) =>
    api.patch<MarketingSocialLink>(`${PREFIX}/social-links/${id}`, body).then((r) => r.data),
  deleteSocial: (id: string) =>
    api.delete(`${PREFIX}/social-links/${id}`).then(() => true),
  reorderSocial: (order: ReorderEntry[]) =>
    api.put(`${PREFIX}/social-links/reorder`, order).then(() => true),

  // Uploads / media library
  listUploads: () =>
    api
      .get<RawUpload[]>(`${PREFIX}/uploads`)
      .then((r) => (r.data ?? []).map(normaliseUpload)),
  uploadFile: (file: File) => {
    const fd = new FormData();
    fd.append('file', file);
    return api
      .post<RawUpload>(`${PREFIX}/uploads`, fd, {
        headers: { 'Content-Type': 'multipart/form-data' },
      })
      .then((r) => normaliseUpload(r.data));
  },
  deleteUpload: (id: string) =>
    api.delete(`${PREFIX}/uploads/${id}`).then(() => true),

  // Launch checklist
  getChecklist: () =>
    api.get<MarketingChecklistItem[]>(`${PREFIX}/launch-checklist`).then((r) => r.data),

  // Industry templates
  listTemplates: () =>
    api.get<MarketingTemplateSummary[]>(`${PREFIX}/templates`).then((r) => r.data),
  getTemplate: (id: string) =>
    api.get<MarketingTemplate>(`${PREFIX}/templates/${id}`).then((r) => r.data),
  useTemplate: (id: string, options: { wipe_existing: boolean }) =>
    api
      .post<MarketingStateOut>(`${PREFIX}/templates/${id}/use`, options)
      .then((r) => r.data),

  // SiteForge JSON export import. Multipart upload of an exported JSON file
  // from SiteForge Studio's Settings → Data → Export. Replaces existing
  // marketing content on success.
  importSiteforge: (file: File) => {
    const form = new FormData();
    form.append('file', file);
    return api
      .post<MarketingStateOut>(`${PREFIX}/import-siteforge`, form, {
        headers: { 'Content-Type': 'multipart/form-data' },
      })
      .then((r) => r.data);
  },
};

/* -------------------------------------------------------------------------- */
/*  Static option lists used by editor UIs.                                    */
/* -------------------------------------------------------------------------- */

export const SECTION_TYPE_OPTIONS: { value: SectionType; label: string; helper: string }[] = [
  { value: 'hero', label: 'Hero', helper: 'Big headline + stats above the fold' },
  { value: 'services', label: 'Services', helper: 'List of what you offer' },
  { value: 'projects', label: 'Projects', helper: 'Portfolio showcase grid' },
  { value: 'testimonials', label: 'Testimonials', helper: 'Social proof quotes' },
  { value: 'about', label: 'About', helper: 'Story / team intro' },
  { value: 'cta', label: 'Call to action', helper: 'Single conversion block' },
];

export const FONT_OPTIONS: string[] = [
  'Inter',
  'Manrope',
  'Plus Jakarta Sans',
  'Space Grotesk',
  'Sora',
  'IBM Plex Sans',
  'Source Sans 3',
  'Roboto',
  'Lora',
  'Playfair Display',
  'Merriweather',
  'JetBrains Mono',
];

export const SOCIAL_PLATFORM_OPTIONS: { value: string; label: string }[] = [
  { value: 'twitter', label: 'X / Twitter' },
  { value: 'linkedin', label: 'LinkedIn' },
  { value: 'github', label: 'GitHub' },
  { value: 'instagram', label: 'Instagram' },
  { value: 'facebook', label: 'Facebook' },
  { value: 'youtube', label: 'YouTube' },
  { value: 'tiktok', label: 'TikTok' },
  { value: 'dribbble', label: 'Dribbble' },
  { value: 'behance', label: 'Behance' },
  { value: 'medium', label: 'Medium' },
  { value: 'mastodon', label: 'Mastodon' },
  { value: 'other', label: 'Other' },
];

/** Derive a URL-safe slug from a plain-text title. */
export function slugify(input: string): string {
  return input
    .toLowerCase()
    .trim()
    .replace(/['"`]/g, '')
    .replace(/[^a-z0-9]+/g, '-')
    .replace(/^-+|-+$/g, '')
    .slice(0, 80);
}

/** Cap-friendly label for an unknown entity type (used in section badges). */
export function sectionTypeLabel(type: string): string {
  const match = SECTION_TYPE_OPTIONS.find((o) => o.value === type);
  return match?.label ?? type;
}
